### PR TITLE
Add %DISPLAYNAME% as parameter and start documenting provisioning

### PR DESCRIPTION
**Pull request by ksteinb**
This pull request adds a %DISPLAYNAME% parameter to the provisioning and tries to document the provisioning a little bit in admin.md

### CODE REVIEW

File: doc/admin.md [hunk 1]
```diff
@@ -25,6 +25,24 @@ You can use the php-mail function to send mails. This is needed for some webhost
 ```php
 'app.mail.transport' => 'php-mail'
 ```
+### Provisioning
+to Provisioning of Mail Accounts for all user navigate to Settings -> Groupware
```
Comment (ChristophWurst) [inline doc/admin.md]: I'm fine with this line, but the rest should be on the UI. Docs here are prone to get out of sync with the UI changes.

File: lib/Service/Provisioning/Config.php [hunk 2]
```diff
@@ -122,6 +122,9 @@ private function buildUserEmail(string $original, IUser $user) {
 		if ($user->getUID() !== null) {
 			$original = str_replace('%USERID%', $user->getUID(), $original);
 		}
+		if ($user->getDisplayName() !== null) {
+			$original = str_replace('%DISPLAYNAME%', $user->getDisplayName(), $original);
```
Comment (ChristophWurst) [inline lib/Service/Provisioning/Config.php]: Please add to https://github.com/nextcloud/mail/blob/df939b93e2993749fbdb43401acedb7cde1adf00/src/components/settings/ProvisioningSettings.vue#L246 as well

**Comment by ksteinb** [comment]
ok, i will change it appropriately, thanks for the hints

File: src/components/settings/ProvisioningSettings.vue [hunk 3]
```diff
@@ -243,7 +243,9 @@
 						/>
 						<br />
 						<small>{{
-							t('mail', "* %USERID% and %EMAIL% will be replaced with the user's UID and email")
+							t('mail', "* %USERID% %DISPLAYNAME and %EMAIL% will be replaced with the user's UID, DISPLAYNAME and email")
```
Comment (ChristophWurst) [inline src/components/settings/ProvisioningSettings.vue]: please remove the two lines above. admins should know the values of the three placeholders and thus be able to figure out what will work for them :)

Comment (ksteinb) [reply src/components/settings/ProvisioningSettings.vue]: is it better now? But please think of not so experienced admins, the documentation of the mail app is really sparse

**Comment by ksteinb** [comment]
is the line too long now ?

File: src/components/settings/ProvisioningSettings.vue (hunk 3 above)
Comment (ChristophWurst) [reply src/components/settings/ProvisioningSettings.vue]: It should be "* %USERID%, %DISPLAYNAME% and %EMAIL% will be replaced with the user's UID, email and display name."

Hm. I'm fine with adding more docs. But they should be written in a form that is comprehensible for everyone and not a listing of edge case recommendations.

Comment (ksteinb) [reply src/components/settings/ProvisioningSettings.vue]: LDAP isn't an edge case in large environments 

Greetings from CS3

Comment (ksteinb) [reply src/components/settings/ProvisioningSettings.vue]: could you tell me where the error in eslint comes from?  I was not yet able to setup the complete build environment locally

Comment (ChristophWurst) [reply src/components/settings/ProvisioningSettings.vue]: > Greetings from CS3

Say hi to Björn and Frank :)

> could you tell me where the error in eslint comes from? I was not yet able to setup the complete build environment locally

ESLint want the `t` call to be wrapped into multiple lines. You can run `npm run lint:autofix` to resolve this automagically.

Comment (ksteinb) [reply src/components/settings/ProvisioningSettings.vue]: Arghh, such a picky thing
i got it now (blind test, as I didn't get eslint running on my Ubuntu LTS 18.04)
but now xml-linters failed, looks like it had connection trouble to apps.nextcloud.com

**Comment by jancborchardt** [comment]
@ksteinb is this ready for review again, or what’s missing? :)

**Comment by ChristophWurst** [comment]
@ksteinb :ping_pong:

**Comment by ChristophWurst** [comment]
closing due to lack of activity, but we can re-open easily if we decide to continue working on the change :)

File: src/components/settings/ProvisioningSettings.vue (hunk 3 above)
Comment (wKollendorf) [reply src/components/settings/ProvisioningSettings.vue]: This should fix it
```
<small>
  {{ t('mail', '* %USERID% %DISPLAYNAME% and %EMAIL% will be replaced with the user\'s UID, display name, and email.') }}<br>
  {{ t('mail', '* In LDAP environments, %USERID% is usually meaningless if IMAP/SMTP login is not possible.') }}<br>
  {{ t('mail', '* With email login, you can work around this by provisioning the display name with the login attribute.') }}
</small>
```

**Pull request closed without merging**